### PR TITLE
LibJS+LibWeb: Implement USVString scalar value handling

### PR DESCRIPTION
**Pull request by shannonbooth**
USVString is defined in the IDL spec as:

> The USVString type corresponds to scalar value strings. Depending on
> the context, these can be treated as sequences of either 16-bit
> unsigned integer code units or scalar values.

This means we need to account for surrogate code points by using the
replacement character.

This fixes the last test in https://wpt.live/url/url-constructor.any.html

### CODE REVIEW

File: Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/IDLGenerators.cpp [hunk 1]
```diff
@@ -332,6 +332,11 @@ static void emit_includes_for_all_imports(auto& interface, auto& generator, bool
 template<typename ParameterType>
 static void generate_to_string(SourceGenerator& scoped_generator, ParameterType const& parameter, bool variadic, bool optional, Optional<ByteString> const& optional_default_value)
 {
+    if (parameter.type->name() == "USVString")
+        scoped_generator.set("to_string", "to_well_formed_string"sv);
+    else
+        scoped_generator.set("to_string", "to_well_formed_string"sv);
```
Comment (skyrising) [inline Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/IDLGenerators.cpp:338]: Shouldn't this be `scoped_generator.set("to_string", "to_string"sv);`? Both of your cases are the same

Comment (shannonbooth) [reply Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/IDLGenerators.cpp:338]: Whoops, yes - unofrtunately i wasnt able to fully test as my laptop was about to die (charging port seems to have broken today). Was working in a reduced test case though w/ all tests pssing.

Just wanted to push up first to get feedback in draft - is this the approach we should be taking for this 🤔 not fully sure yet

**Comment by shannonbooth** [comment]
Should be good to go now I think

**Comment by shannonbooth** [comment]
(from a new laptop 😸 )

**Review by tcl3 — APPROVED** [review]

**Pull request merged**